create-release.sh

```diff
@@ -0,0 +1,293 @@
+          continue
+        fi
+        break
+        ;;
+      *)
+        echo "Invalid selection. Please enter a number between 1 and 4."
+        ;;
+    esac
+  done
+}
+
+# Validate arguments
+if [[ -z "$VERSION_TYPE" && -z "$CUSTOM_VERSION" ]]; then
+  # No arguments provided, use interactive menu
+  select_version_type
+fi
+
+if [[ -n "$VERSION_TYPE" && -n "$CUSTOM_VERSION" ]]; then
+  echo "Error: You cannot specify both a version type and a custom version"
+  exit 1
+fi
+
+# Get current version from package.json
+CURRENT_VERSION=$(node -e "console.log(require('./package.json').version)")
+echo "Current version: $CURRENT_VERSION"
+
+# Calculate new version
+if [[ -n "$CUSTOM_VERSION" ]]; then
+  NEW_VERSION="$CUSTOM_VERSION"
+else
+  # Split the version into components
+  IFS='.' read -r MAJOR MINOR PATCH <<< "$CURRENT_VERSION"
+
+  if [[ "$VERSION_TYPE" == "patch" ]]; then
+    NEW_VERSION="${MAJOR}.${MINOR}.$((PATCH + 1))"
+  elif [[ "$VERSION_TYPE" == "minor" ]]; then
+    NEW_VERSION="${MAJOR}.$((MINOR + 1)).0"
+  elif [[ "$VERSION_TYPE" == "major" ]]; then
+    NEW_VERSION="$((MAJOR + 1)).0.0"
+  fi
+fi
+
+echo "New version: $NEW_VERSION"
+
+# Find the upstream remote
+UPSTREAM_REMOTE=$(find_mui_x_remote)
+echo "Found upstream remote: $UPSTREAM_REMOTE"
+
+# Update the upstream master branch
+echo "Updating the upstream master branch..."
+git fetch $UPSTREAM_REMOTE master
+
+# Check for uncommitted changes
+if [ -n "$(git status --porcelain)" ]; then
+  echo "Error: You have uncommitted changes. Please commit or stash them before running this script."
+  exit 1
+fi
+
+# Create a new branch with the new version
+BRANCH_NAME="release/v$NEW_VERSION"
+echo "Creating new branch: $BRANCH_NAME"
+git checkout -b $BRANCH_NAME $UPSTREAM_REMOTE/master
+
+# Update package.json
+echo "Updating package.json..."
+# Use node to update the package.json file
+node -e "
+const fs = require('fs');
+const packageJson = require('./package.json');
+packageJson.version = '$NEW_VERSION';
+fs.writeFileSync('./package.json', JSON.stringify(packageJson, null, 2) + '\n');
+"
+
+echo "Updated package.json version to $NEW_VERSION"
+
+# Run lerna version script
+echo "Running lerna version script..."
+npx lerna version --exact --no-changelog --no-push --no-git-tag-version --no-private
+
+echo "Version update completed successfully!"
+echo "New version: $NEW_VERSION"
+
+# Generate the changelog
+echo "Generating changelog..."
+CHANGELOG_CONTENT=$(node scripts/releaseChangelog.mjs --githubToken=$GITHUB_TOKEN --nextVersion=$NEW_VERSION --returnEntry)
+
+# Add the new changelog entry to the CHANGELOG.md file
+echo "Adding changelog entry to CHANGELOG.md..."
+# Find the position of the first version entry (currently ## 8.5.1)
+FIRST_VERSION_LINE=$(grep -n "^## [0-9]" CHANGELOG.md | head -1 | cut -d: -f1)
+
+# Create a temporary file with the new content
+head -n 7 CHANGELOG.md > temp_changelog.md  # Keep the header (first 7 lines)
```

Comment:
```suggestion
head -n $(($FIRST_VERSION_LINE-1)) CHANGELOG.md > temp_changelog.md
```

Might as well use the calculated value in case we change the header